### DISCUSSION TITLE
Lesson 6 - brownie test vs python pytest

### DISCUSSION THREAD

**Discussion by 20skcs** [smartcontractkit/full-blockchain-solidity-course-py#915] [Q&A] [answered]
This is my first time submitting a question here so my apologies in advance if I don't properly follow normal practice in how this is presented. My question (with details provided below) is whether there is any significant/important difference in the testing process completed by running 'python -m pytest' as opposed to 'brownie test'? Brownie test does not seem to run for me as it does in the tutorial (details below).

When I attempt to run 'brownie test -k test_fund_me.py' 0 items are collected and no tests are ran.

When I run 'brownie test', I receive the same result (0 items collected).

When I run 'python -m pytest', the test completes successfully.

I am running windows 10.

Thank you in advance for any help you can provide.

```


PS C:\Users\robyn\Demos\brownie_fund_me> brownie test -k test_fund_me.py                 
INFO: Could not find files for the given pattern(s).
Brownie v1.16.4 - Python development framework for Ethereum

==================================================================== test session starts ====================================================================
platform win32 -- Python 3.10.2, pytest-6.2.5, py-1.10.0, pluggy-1.0.0
rootdir: C:\Users\robyn\Demos\brownie_fund_me
plugins: eth-brownie-1.16.4, hypothesis-6.21.6, forked-1.3.0, xdist-1.34.0, web3-5.23.1
collected 0 items

=================================================================== no tests ran in 0.02s ===================================================================
PS C:\Users\robyn\Demos\brownie_fund_me> brownie test -k test_fund_me.py



PS C:\Users\robyn\Demos\brownie_fund_me> python -m pytest
INFO: Could not find files for the given pattern(s).
==================================================================== test session starts ====================================================================
platform win32 -- Python 3.10.2, pytest-6.2.5, py-1.10.0, pluggy-1.0.0
rootdir: C:\Users\robyn\Demos\brownie_fund_me
plugins: eth-brownie-1.16.4, hypothesis-6.21.6, forked-1.3.0, xdist-1.34.0, web3-5.23.1
collected 2 items

Launching 'ganache-cli.cmd --accounts 10 --hardfork istanbul --gasLimit 12000000 --mnemonic brownie --port 8545'...

contracts\test\test_fund_me.py ..                                                                                                                      [100%]

==================================================================== 2 passed in 10.73s ===================================================================== 
Terminating local RPC client...


```

**Comment by cromewar** [comment, marked as answer]
Hello @20skcs, could you please share a screenshot of the folder structure on your project?
If I have to guess I would say the test scripts are on the incorrect folder so brownie is unable to find them.

**Reply by 20skcs** [reply]
Thank you very much for your reply and for your suggestion on the possible issue. Your answer was spot on, that was exactly what the problem was. For some reason I had created the test folder (and the test file) within the contracts folder rather than creating the test file in the brownie_fund_me/tests folder.

**Reply by cromewar** [reply]
@20skcs I'm glad you solved it, feel free to ask any other question.
